.github/workflows/sycl-detect-changes.yml

```diff
@@ -22,13 +22,13 @@ jobs:
         uses: actions/github-script@v7
         with:
           script: |
-            console.log("Number of files changed:");
-            console.log(context.payload.pull_request.changed_files);
-            return context.payload.pull_request.changed_files ;
+            const changedFiles = context.payload.pull_request.changed_files;
```

Comment:
"For numerical comparison, the fromJSON() function can be used to convert a string to a number. For more information on the fromJSON() function, see fromJSON."

https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions

Can we just use this in the `if`?

Reply:
I don't think we can. My understanding is that `if` in steps is not Javascript. We can only access `context` only in `JavaScript` ( script in steps).

"The script directly accesses
context.payload.pull_request.changed_files and performs the comparison
within the JavaScript environment, which is more flexible and allows for
direct manipulation of data."

"The if condition in GitHub Actions is evaluated using the expression
syntax, which requires correct access to step outputs and proper
handling of data types. "